cpp/include/cudf/interop.hpp

```diff
@@ -84,15 +84,20 @@ DLManagedTensor* to_dlpack(
  */
 
 /**
- * @brief Detailed meta data information for arrow array.
+ * @brief Detailed metadata information for arrow array.
  *
  * As of now this contains only name in the hierarchy of children of cudf column,
  * but in future this can be updated as per requirement.
  */
 struct column_metadata {
-  std::string name;
-  std::vector<column_metadata> children_meta;
+  std::string name;                            ///< Name of the column
+  std::vector<column_metadata> children_meta;  ///< Metadata of children of the column
 
+  /**
+   * @brief Construct a new column metadata object
+   *
+   * @param _name Name of the column
+   */
   column_metadata(std::string const& _name) : name(_name) {}
```

Comment:
I am aware of this. But in the past, many reviews suggested to have different name for constructor parameters or member variables to differentiate. IIRC @jrhemstad @devavret @cwharris were involved in those reviews. So, this suggestion could use some retrospection. 

This might be coming from python convention @vyasr  https://github.com/rapidsai/cudf/pull/9889#discussion_r771486273

Reply:
Using underscores in specific ways is certainly a Python convention, but there's no real equivalent this syntax since there are no initializer lists and no implicit member accesses without `self` in Python. I'm not sure what previous discussions were had indicating a particular style preference, though.